### ISSUE TITLE
Incorrect MSRV due to error parsing Cargo.lock with older toolchain

### ISSUE THREAD

**Issue by staticintlucas** [foresterre/cargo-msrv#73]
One of my projects is supported by Rust 1.30.0+, however cargo-msrv reports that the minimum version is 1.38.0.

If I manually run `cargo check` or `cargo test` with an earlier version:

```text
$ cargo +1.37.0 check
error: failed to parse lock file at: /home/lucas/stringslice/Cargo.lock

Caused by:
  invalid serialized PackageId for key `package.dependencies`
```

It looks like the format of Cargo.lock has changed slightly between 1.37.0 and 1.38.0. If I delete Cargo.lock and rerun `cargo check` or `cargo test` it passes:

```text
$ rm Cargo.lock
$ cargo +1.37.0 check
    Updating crates.io index
    Finished dev [unoptimized + debuginfo] target(s) in 2.94s
```

However, if I try this with `cargo msrv` it will first regenerate the Cargo.lock file with the latest stable Rust (currently 1.52.0), which then fails to parse when it gets to 1.37.0.

The only way I can get it to correctly report the correct MSRV is running with `--maximum=1.37.0` so that it generates a Cargo.toml compatible with Rust<=1.37.0:

```text
$ rm Cargo.lock
$ cargo msrv --maximum=1.37.0 --minimum=1.20.0 -- cargo test -- --test-threads=1
Determining the Minimum Supported Rust Version (MSRV) for toolchain x86_64-unknown-linux-gnu
Using check command cargo test -- --test-threads=1
   Finished The MSRV is 1.30.1
████████████████████████████████████████████████████████████████████ 00:00:28
```

The best solution I can think of is adding a `--ignore-lock-file` flag to move/delete Cargo.lock and let Cargo regenerate it with each toolchain version. This means gives the user the option of whether to use or ignore the lock file.

**Comment by foresterre** [comment]
Thank you for the detailed bug report :smile:!

I aim to fix this in the next a few days.

**Comment by foresterre** [comment]
I implemented this in #75. If you have time, and are willing, could you try whether it works for you, thank you 😄 ?

*Steps:*
- Install (update): `cargo install cargo-msrv` 
- Run: `cargo msrv --version` (should be 0.5.0)
- Run `cargo msrv --ignore-lockfile`

Alternatively, you can replace the Install step with the following:
- `cargo install cargo-msrv --git https://github.com/foresterre/cargo-msrv.git --branch main`

**Comment by staticintlucas** [comment]
I tested the new version and I can confirm it works 🎉 

Thanks for the quick fix!
